### PR TITLE
feat(ConfigUseCase): log AppCheck failures

### PR DESCRIPTION
**Pull request by KaylaBrady**
### Summary

An action item from _Ticket:_ [Fix "map tile crash"](https://app.asana.com/0/1205732265579288/1208436806470885/f)

What is this PR for?

We have a plausible fix for the map tile crash in https://github.com/mbta/mobile_app/pull/464, but the investigation of that ticket also considered whether an AppCheck failure could be at play. This ticket adds logging for AppCheck failures so that we have more insight into whether / how frequently those occur.

### Testing

What testing have you done?

Added unit tests and triggered locally. Opted to create a SentryRepository to simplify testability.

Also confirmed captured[ in Sentry](https://mbtace.sentry.io/issues/5985487155/?environment=debug&project=4506321052172288&referrer=project-issue-stream)

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/usecases/ConfigUseCase.kt [hunk 1]
```diff
@@ -4,23 +4,31 @@ import com.mbta.tid.mbta_app.model.response.ApiResult
...
+                        "AppCheck token error ${tokenResult.code} ${tokenResult.message}"
+                    )
                     ApiResult.Error(message = "app check token failure ${tokenResult.message}")
+                }
                 is ApiResult.Ok -> {
                     configRepo.getConfig(tokenResult.data.token)
                 }
             }
         } catch (e: Exception) {
+            Sentry.captureException(e)
```
Comment (boringcactus) [inline shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/usecases/ConfigUseCase.kt:31]: Should this be sent through the repo as well?

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/dependencyInjection/RepositoryDI.kt [hunk 2]
```diff
@@ -82,6 +86,7 @@ class RepositoryDI : IRepositories, KoinComponent {
     override val schedules: ISchedulesRepository by inject()
     override val searchResults: ISearchResultRepository by inject()
     override val settings: ISettingsRepository by inject()
+    override val sentry: ISentryRepository by inject()
```
Comment (boringcactus) [inline shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/dependencyInjection/RepositoryDI.kt:89]: `sentry` should be before `settings` here. (Android Studio has a handy "Edit > Sort Lines" menu option.)

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/endToEnd/EndToEndRepositories.kt [hunk 3]
```diff
@@ -167,6 +167,7 @@ fun endToEndModule(): Module {
         }
         single<ISearchResultRepository> { MockSearchResultRepository() }
         single<ISettingsRepository> { MockSettingsRepository() }
+        single<ISettingsRepository> { MockSettingsRepository() }
```
Comment (boringcactus) [inline shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/endToEnd/EndToEndRepositories.kt:170]:
```suggestion
        single<ISentryRepository> { MockSentryRepository() }
        single<ISettingsRepository> { MockSettingsRepository() }
```

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/dependencyInjection/RepositoryDI.kt (hunk 2 above)
Comment (KaylaBrady) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/dependencyInjection/RepositoryDI.kt:89]: Oh cool, thanks for the tip!

**Review by boringcactus — APPROVED** [review]

**Pull request merged**